library/Rules/Lazy.php

```diff
@@ -21,22 +21,26 @@
 #[Attribute(Attribute::TARGET_PROPERTY | Attribute::IS_REPEATABLE)]
 final class Lazy extends Standard
 {
-    /** @var callable(mixed): Rule */
+    /** @var callable(mixed): Rule|callable(mixed): Result */
     private $ruleCreator;
 
-    /** @param callable(mixed): Rule $ruleCreator */
+    /** @param callable(mixed): Rule|callable(mixed): Result $ruleCreator */
```

Comment:
Gotta move that to a different commit, it's not part of those changes, but a much needed change nonetheless